### PR TITLE
Add support for regex string for password policy validatorPattern setting

### PR DESCRIPTION
**Pull request by bhaskaryasa**
fixes issue #3302

### CODE REVIEW

File: spec/PasswordPolicy.spec.js [hunk 1]
```diff
@@ -231,6 +253,40 @@ describe("Password Policy: ", () => {
...
+    const user = new Parse.User();
+    reconfigureServer({
+      appName: 'passwordPolicy',
+      passwordPolicy: {
+        validatorPattern: "[!@#$]+"  // password should contain at least one special char
+      },
+      publicServerURL: "http://localhost:8378/1"
+    }).then(() => {
+      user.setUsername("user1");
+      user.setPassword("p@sswrod");
```
Comment (acinader) [inline spec/PasswordPolicy.spec.js]: misspelled password :)....

**Comment by acinader** [comment]
@bhaskaryasa this looks good to me.  one question, if you supply a bad regex string like say: `\xx`

what happens?  I'd think that you'd want the server to not start and make sure that you get a decent error message or stack trace so it is obvious what is causing the failure?  

what do you think?

**Comment by livingstonef** [comment]
@bhaskaryasa This is good! Particularly thorny issue if you are using the parse-server lib directly (i.e $ parse-server /path-to-config.json). It is basically impossible to supply either a validation callback or pure regex. this fixes the regex bit. still don't think you can pass a callback in the currently documented way.

@acinader before this fix, if you submit a regex string it fails JSON validation with "unexpected token..."

**Comment by flovilmart** [comment]
@livingstonef you can provide a 'js' config file as well, as it's loaded through a `require` call :)

**Comment by bhaskaryasa** [comment]
@acinader - I believe JSON parsing itself would fail if you passed something like '\xx'. That should cause the Parse Server startup failure with the exception logged.

**Comment by bhaskaryasa** [comment]
@flovilmart - I did not know that was possible!! Nice 👍

**Comment by acinader** [comment]
@bhaskaryasa yeah, that'd be my guess to.  Can you please verify that the server doesn't start and that the error doesn't get eaten somehow?

**Comment by bhaskaryasa** [comment]
@acinader - Yes, tested. Startup fails with SyntaxError: Unexpected token x in JSON at position 202.
I think we are good.

**Comment by dkarviga** [comment]
@flovilmart could you please provide an example of js file that can be used for initialization of parse-server?

**Comment by flovilmart** [comment]
```
// config.js
module.exports = {
  appId: "hello",
  masterKey: "world"
  /* more options */
}
```

parse-server ./config.js

**Comment by dkarviga** [comment]
@flovilmart thanks!

**Pull request merged**